*** squirrelmail-1.2.7-original/functions/mailbox_display.php	Sun Aug 25 21:42:26 2002
--- squirrelmail-1.2.7/functions/mailbox_display.php	Mon Aug 26 05:32:42 2002
***************
*** 151,157 ****
  	  . $msg["ID"]."$checked></TD>\n";
  	break;
        case 2: /* from */
! 	echo "   <td bgcolor=\"$hlt_color\">$italic$bold$flag$fontstr"
  	  . "$senderName$fontstr_end$flag_end$bold_end$italic_end</td>\n";
  	break;
        case 3: /* date */
--- 151,159 ----
  	  . $msg["ID"]."$checked></TD>\n";
  	break;
        case 2: /* from */
! 	echo "   <td bgcolor=\"$hlt_color\" valign=\"middle\">";
!         do_hook("message_icon", $msg['FROM'], 0);
! 	echo "$italic$bold$flag$fontstr"
  	  . "$senderName$fontstr_end$flag_end$bold_end$italic_end</td>\n";
  	break;
        case 3: /* date */
*** squirrelmail-1.2.7-original/src/read_body.php	Sun Aug 25 21:42:27 2002
--- squirrelmail-1.2.7/src/read_body.php	Mon Aug 26 05:36:34 2002
***************
*** 802,807 ****
--- 802,810 ----
         '<TR><TD WIDTH="100%">' .
         '<TABLE WIDTH="100%" BORDER="0" CELLSPACING="0" CELLPADDING="3">' . "\n" .
            '<TR>' . "\n";
+     echo "         <TD BGCOLOR=\"$color[0]\" ROWSPAN=\"5\" ALIGN=\"RIGHT\" VALIGN=\"MIDDLE\">\n";
+     do_hook("message_icon", $from_name, 1);
+     echo "         </TD>";
  
  /** subject **/
  echo          "<TD BGCOLOR=\"$color[0]\" WIDTH=\"10%\" ALIGN=\"right\" VALIGN=\"top\">\n" .
